gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -299,31 +299,16 @@ end
 -- Adds an @{ITEM} or a @{Weapon} into the bought list of a @{Player}
 -- @note This will disable another purchase of the same equipment
 -- if this equipment is limited
--- @param string cls
+-- @param string equipmentId
 -- @realm server
 -- @see Player:RemoveBought
-function plymeta:AddBought(cls)
+function plymeta:AddBought(equipmentId)
 	self.bought = self.bought or {}
-	self.bought[#self.bought + 1] = tostring(cls)
-
-	BUYTABLE[cls] = true
-
-	net.Start("TTT2ReceiveGBEq")
-	net.WriteString(cls)
-	net.Broadcast()
+	self.bought[#self.bought + 1] = tostring(equipmentId)
 
-	local team = self:GetTeam()
-
-	if team and team ~= TEAM_NONE and not TEAMS[team].alone then
-		TEAMBUYTABLE[team] = TEAMBUYTABLE[team] or {}
-		TEAMBUYTABLE[team][cls] = true
-
-		if SERVER then
-			net.Start("TTT2ReceiveTBEq")
-			net.WriteString(cls)
-			net.Send(GetTeamFilter(team))
-		end
-	end
+	shop.SetEquipmentBought(self, equipmentId)
+	shop.SetEquipmentGlobalBought(equipmentId)
+	shop.SetEquipmentTeamBought(self, equipmentId)
```

Comment:
Tbh I don't like the design of this api.
This rather should be done by a single call. 

Also i don't think its nice to have 3 tables for something that just derives from a single one. Because team and global limitations can be checked by checking the bought table.

I know this is how we did it before and not a concept of yours but nonetheless i propose that we try to get rid of this and make it easier

Reply:
I'll see what I can do ^^

Reply:
Nah, lets do it in another PR.